components/ArchivePosts.php

```diff
@@ -0,0 +1,205 @@
+            ],
+            'noPostsMessage' => [
+                'title'        => 'rainlab.blog::lang.settings.posts_no_posts',
+                'description'  => 'rainlab.blog::lang.settings.posts_no_posts_description',
+                'type'         => 'string',
+                'default'      => 'No posts found',
+                'showExternalParam' => false
+            ],
+            'sortOrder' => [
+                'title'       => 'rainlab.blog::lang.settings.posts_order',
+                'description' => 'rainlab.blog::lang.settings.posts_order_description',
+                'type'        => 'dropdown',
+                'default'     => 'published_at desc'
+            ],
+            'categoryPage' => [
+                'title'       => 'rainlab.blog::lang.settings.posts_category',
+                'description' => 'rainlab.blog::lang.settings.posts_category_description',
+                'type'        => 'dropdown',
+                'default'     => 'blog/category',
+                'group'       => 'Links',
+            ],
+            'postPage' => [
+                'title'       => 'rainlab.blog::lang.settings.posts_post',
+                'description' => 'rainlab.blog::lang.settings.posts_post_description',
+                'type'        => 'dropdown',
+                'default'     => 'blog/post',
+                'group'       => 'Links',
+            ],
+        ];
+    }
+
+    public function getCategoryPageOptions()
+    {
+        return Page::sortBy('baseFileName')->lists('baseFileName', 'baseFileName');
+    }
+
+    public function getPostPageOptions()
+    {
+        return Page::sortBy('baseFileName')->lists('baseFileName', 'baseFileName');
+    }
+
+    public function getSortOrderOptions()
+    {
+        return BlogPost::$allowedSortingOptions;
+    }
+
+    public function onRun()
+    {
+        $this->prepareVars();
+        $this->posts = $this->page['posts'] = $this->listPosts($this->property('slug'));
+
+        /*
+         * If the page number is not valid, redirect
+         */
+        if ($pageNumberParam = $this->paramName('pageNumber')) {
+            $currentPage = $this->property('pageNumber');
+
+            if ($currentPage > ($lastPage = $this->posts->lastPage()) && $currentPage > 1)
+                return Redirect::to($this->currentPageUrl([$pageNumberParam => $lastPage]));
+        }
+    }
+
+    protected function prepareVars()
+    {
+        $this->pageParam = $this->page['pageParam'] = $this->paramName('pageNumber');
+        $this->noPostsMessage = $this->page['noPostsMessage'] = $this->property('noPostsMessage');
+
+        /*
+         * Page links
+         */
+        $this->postPage = $this->page['postPage'] = $this->property('postPage');
+        $this->categoryPage = $this->page['categoryPage'] = $this->property('categoryPage');
+    }
+
+    protected function listPosts($slug)
+    {
+        try {
+            $slug = str_replace('+', ' ', $slug);

```

Comment:
What's the rationale behind using `str_replace` instead of `urldecode`? http://php.net/manual/en/function.urldecode.php?